src/auth/activation.ts

```diff
@@ -48,8 +51,74 @@ export async function initialize(
  */
 async function showManageConnectionsOnStartup() {
     if (!ExtensionUse.instance.isFirstUse()) {
+        getLogger().debug(
+            'firstStartup: This is not the users first use of the extension, skipping showing Add Connections page.'
+        )
+        return
+    }
+
+    if (hasExistingConnections()) {
+        // Just in case isFirstUse() is incorrect, but they have connections, they probably aren't new
+        getLogger().debug('firstStartup: The user has existing connections, skipping showing Add Connections page.')
+        return
+    }
+
+    if (isInDevEnv()) {
+        // A dev env will have an existing connection so this scenario is redundant. But keeping
+        // for reference.
+        getLogger().debug('firstStartup: Detected we are in Dev Env, skipping showing Add Connections page.')
         return
     }
 
-    AuthCommandDeclarations.instance.declared.showConnectionsPage.execute()
+    AuthCommandDeclarations.instance.declared.showConnectionsPage.execute('firstStartup')
+    emitFirstStartupMetrics()
+}
+
+/**
+ * Return true if the user has existing connections that
+ * the extension is aware of.
+ */
+function hasExistingConnections(): boolean {
+    /**
+     * This specific property/function does not search for user credentials
+     * on their local machine, it looks at the current values in its internal store.
+     * So if this returns false it simply means the extension is not aware of it yet,
+     * but credentials could exist in something like a `credentials` file.
+     *
+     * If the user has existing credentials on their system, but this returns false,
+     * we can assume this extension has not run before since it would
+     * have discovered them and added to the extensions internal store, resulting in
+     * this returning true.
+     */
+    return Auth.instance.hasConnections
+}
+
+async function emitFirstStartupMetrics() {
+    // HACK: Telemetry client may not be initialized yet, wait until it exists
+    await waitUntil(async () => telemetry, {
+        interval: 500,
+        timeout: 30000,
+    })
+
+    // Metric that is emitted for ALL new users
+    telemetry.auth_addConnection.emit({
+        source: 'firstStartup' as AuthSource,
+        reason: 'firstStartup',
+        result: 'Cancelled',
+    })
+
+    // Metrics that are emitted if existing connections are detected
+    const allConnections = await Auth.instance.listConnections() // implicitly loads user credentials
+    const reason = 'alreadyHadAuth'
+
+    const credentialsConnections = allConnections.filter(isIamConnection)
+    if (credentialsConnections.length > 0) {
+        telemetry.auth_addConnection.emit({
+            source: 'firstStartup' as AuthSource,
+            reason,
```

Comment:
The user might not have added the connection through the Toolkit though? Emitting this event is misleading.

Reply:
Removed this, not needed anymore